### ISSUE TITLE
APT Weekly Check-in

### ISSUE THREAD

**Issue by wrynearson** [NASA-IMPACT/nasa-apt#556]
This ticket will be used for weekly planning and check-ins to partially replace weekly meetings, mostly due to the time zone difference between the DevSeed and IMPACT teams.

@NASA-IMPACT/apt

**Comment by oliverroick** [comment]
- Version changelog (https://github.com/NASA-IMPACT/nasa-apt/issues/527) is implemented and merged, but still needs to be deployed
- Thread mentions (https://github.com/NASA-IMPACT/nasa-apt/issues/456#issuecomment-1216550387).
  - Notifying individual users is implemented and merged but. not yet deployed
  - Currently working through notifying curators, will wrap this this week
- Will look into bug raised for the sign-up process (https://github.com/NASA-IMPACT/nasa-apt/issues/549)

**Comment by deborahUAH** [comment]
- I cancelled standing Wednesday meetings and created Monday 10am CT meetings for week plan/scrum (I culled the invite list,  let me know if I removed anyone who should be added back)
- On Thursdays, all team members are expected to note progress here.  Demos as needed for biweekly reviews
- Checked in with Brad and Shawn to ensure everyone knows the new meeting cadence and has plans for the week
- I am finishing user guide and UI content so that next sprint changes can be implemented
- Review is this Friday.  APT slide will be ready by 5pm PT on Thursday. Please review by Friday morning CT and make any needed changes
- Aaron is adjust text of cognito #549 ,  Brad is recreating the demo #552  and I am completing the user guide #472  to remove the block

**Comment by oliverroick** [comment]
- Investigated bug with feedback form (https://github.com/NASA-IMPACT/nasa-apt/issues/557); needs whitelisting URLs on IMPACT's side
- Implemented notifying and highlighting curators (https://github.com/NASA-IMPACT/nasa-apt/issues/456#issuecomment-1216550387); currently in code review

**Comment by deborahUAH** [comment]
APT review slide completed. Please see if I missed anything or misrepresented anything.  https://docs.google.com/presentation/d/1N2o8uh3bl8zEMELoLbfLbRKD53UwVa_vHZri-gD6gK8/edit#slide=id.g10e16231a05_0_8

**Comment by wrynearson** [comment]
Hi @deborahUAH , could you help us set up a meeting with someone at ESDIS, or is there someone you can point us to? We have multiple questions about security requirements.

Cc: @aboydnw

**Comment by wrynearson** [comment]
## Mid-week check-in (Aug 22-25):
- Pod week for the APT team with a focus on team building and internal DevSeed things, so less feature development.
- Deployed OpenSearch
    - Document search now works on the Dashboard page
    - Still debugging letter-by-letter search

**Comment by deborahUAH** [comment]
@wrynearson @aboydnw  I will make time to arrange a meeting next week for either later in the week or the week after. Is that quick enough?

**Comment by wrynearson** [comment]
Thanks @deborahUAH, that's great. I won't be available on Wednesday and none of us will be available on Thursday. Could you try for Tuesday, Sep 6th in your morning?

We'll have SNWG questions as well, cc: @TimMcCauley @oliverroick

**Comment by wrynearson** [comment]
## Mid-week update:

(sorry for the late update – my Thursday afternoon was filled with back-to-back meetings)

- For the HTTP(S) discussion regarding the APIs, these are already HTTPS, so we don't need to change anything. The name of the AWS service says HTTP, but it's really HTTPS.
- Thread mentions are finished, but have not been deployed to production.
- For change logs, from [this list](https://github.com/NASA-IMPACT/nasa-apt/issues/527#issuecomment-1224590461), 1 and 3 are complete, while 2 is close to completion.
- @naomatheus is back in the US 🥳 We'll work with him for next steps regarding OpenSearch
- Allow downloading PDF from change log modals (not sure if this is deployed)

As far as I know, there is nothing to demo.

**Comment by deborahUAH** [comment]
Thanks @wrynearson for the update.  Monday is a US holiday.  We meet Wednesday at 10 am CT to discuss sprint work and FY23 plans.  Talk to you then.  Reminder:  Friday is my last day before vacation (gone 3 weeks), Aaron will be covering as lead of project.

**Comment by wrynearson** [comment]
## Personal update:

Somehow it slipped my mind to mention that I'll be out this Friday (Sep 9) and the following Monday (Sep 12) to move to another apartment.

As I mentioned in the call yesterday, I'll be out from October 3rd and will be back on October 18th.

**Comment by wrynearson** [comment]
## Mid-week update:
- #500 is proving to be a big fix.
    - When we tested OpenSearch, the deployment failed, but so too did the rollback, which was done automatically by AWS.
    - Our main developer working on this, @naomatheus, was out intermittently until today.
    - Tl;dr, we need to do much more testing locally before redeploying to staging. We expect to have this done by the end of next week (9/16).
- #527 is merged and is live on staging. @deborahUAH , please review this.
- #456 is mostly ready, but we're waiting on database migrations related to #500 fixes
- #480 is merged and is live on staging. @deborahUAH , please review this.
- @naomatheus is looking into [the issue](https://github.com/NASA-IMPACT/nasa-apt/issues/545#issuecomment-1235904742) of not being to generate PDFs.
- I wrote up #578

Staging accounts also needed to be reset due to the issue with #500 (above).
- Dummy accounts: reviewer@apt.com, contributor@apt.com, contributor2@apt.com, and curator@apt.com all share the same password: `Password123!`
- Your personal accounts (anthonyboyd@developmentseed.org, bwb0013@apt.com, oliver@developmentseed.org, matthew@@developmentseed.org, will@developmentseed.org, timothy@developmentseed.org, and aaron.s.kaulfus@nasa.gov) have the default password `vnYCpsi!j8c6`, which you can change.
- Unfortunately, this process will probably repeat for staging next week.

@deborahUAH , are you able to approve my APT account will@developmentseed.org in production, or is someone else in control of that?

**Comment by wrynearson** [comment]
# Mid-week update:
- Addressed some big fixes and feedback re change logs
- Wrote tests for publishing workflow, opened PR for first batch of tests
- Made a new static that redirects from the incorrect to correct site, but I'm waiting to hear when it will be deployed
- Still working on OpenSearch. The new best estimate is that this will be working locally (for the rest of the DevSeed team to check) by the end of the week, and will be implemented on staging by mid-next week.

**Comment by wrynearson** [comment]
## Mid-week update:
- local opensearch working, branch can be pulled to test locally for others
- working on deploying a dev backend with opensearch (we already have one without opensearch and elasticsearch was removed)
- @TimMcCauley will review @oliverroick 's various PRs early next week. We will merge them separately from @naomatheus 's OpenSearch work.
- I've been planning some PI prep work, and will share it at the meeting on Monday

**Comment by wrynearson** [comment]
## Mid-week update:

- OpenSearch, thread mentions, journal publication, and version change logs are in staging for testing. Awaiting approval before deploying to production
- @naomatheus is close to getting his badge, then he can start on production deployments.

**Comment by wrynearson** [comment]
## Mid-week update:

- Matt is set up for MCP deployments 🎆 
- He’ll deploy Oliver’s PR (https://github.com/NASA-IMPACT/nasa-apt-frontend/pull/397) Thursday morning (today) to production
- He’ll then deploy #500 to production
- He is is testing #545 and will reach out if he has any questions. We expect this to be resolved next week.

**Comment by deborahUAH** [comment]
Thanks, Will.   Glad to hear!   I have added to ticket #586 per user feedback.  
The meeting with ESDIS went well.  @naomatheus did a great job responding to questions and providing information.  Couldn't have had the meeting without him.
Please check slides for Friday AM (CT) review.  Adjust as needed.  https://docs.google.com/presentation/d/1z0TCNVISxEWx-CGnICgLCuffLPphX8R4rv-_QIPSTTY/edit#slide=id.g10e16231a05_0_8

**Comment by wrynearson** [comment]
Thanks @deborahUAH , great to hear the meeting with ESDIS went well. Thanks @naomatheus for being there!

I don't have access to the slides, so I won't be able to review them before the sprint review session.

**Comment by wrynearson** [comment]
## Mid-week update

- #500 is deployed
- https://github.com/NASA-IMPACT/nasa-apt-frontend/pull/397 is deployed
- #545 is on its way
- #557 is live on staging, can be pushed to Prod when approved
- Confirmed flow of #548

**Comment by wrynearson** [comment]
## Mid-week update:
- #545 is in internal review, should be deployed to staging early next week.
- #557 will be pushed to prod, we're just checking that it doesn't conflict with pushing #545 

If I missed something, please feel free to add it here @naomatheus

**Comment by wrynearson** [comment]
## Mid-week update:
- Reviewing feedback from #545 
    - Incorporating new bug, #593 
- #557 pushed to production.

**Comment by wrynearson** [comment]
## Mid-week update:

Happy Thanksgiving!

- Nov 21-25 was a flex week for DevSeed, so no visible updates.
- Short week for @naomatheus for 🦃 .
- Progress is being made on #593 
- Thanks @deborahUAH for the email regarding the document lock not being cleared when it should be. I wrote it up in #594. Please add this to the project board. To me, this seems like a low-priority bug, because the user is "over-warned" rather than "under-warned", but we can discuss this.

**Comment by wrynearson** [comment]
## Mid-week update:
- #593 is in review: https://github.com/NASA-IMPACT/nasa-apt/pull/595
- Image and caption placeholder text is in review (now a backup for ☝️)
- @naomatheus will next look at #583 and #594 (not originally planned for this sprint).

**Comment by naomatheus** [comment]
### Update (APT):
- Work on #595 and #583 is consolidated in [PR 595](https://github.com/NASA-IMPACT/nasa-apt/pull/595).
- I've requested for latest updates to #595 to be reviewed and pushed to staging after approval from @wrynearson and another engineer.
- PR 595 will resolve most PDF expected functions and is being reviewed to ensure that text formatting (bold, italic) can be used properly in generated PDFs.
- Regarding DOIs: [APT Frontend 400](https://github.com/NASA-IMPACT/nasa-apt-frontend/pull/400), I've asked for a frontend review to ensure nothing breaks when pushing this to staging.

FYI:
I am off 12/9 and will continue with priorities 12/12.

**Comment by wrynearson** [comment]
## Mid-week update
- #622 
    - Great progress – we have the new rearchitected approach live on staging (info [here](https://github.com/NASA-IMPACT/nasa-apt/issues/622#issuecomment-1413995420) for how to access it)
    - Setting up the infrastructure to let users download it through the download button is close (it was more complex than we anticipated)
    - Table of contents and line breaks are the next work items
    - Styling changes will now take _far_ less time
- #608
    - Research has been done, some prep work implemented
    - MFA has been turned on (optional, authenticator app), we'll test that tomorrow and will update for next steps on the ticket ☝️

**Comment by wrynearson** [comment]
## Mid-week update
- #622
    - Finishing up work on authentication, which will let us use the current download button to download the new PDFs
    - Assessing styling needs, ToC, and line breaks
    - "Future-proofing" this solution to make it work efficiently and consistently (fast, reliable downloads)

cc: @batpad

**Comment by wrynearson** [comment]
## Mid-week update
- #608 is on staging. @xhagrg's requested changes will be added later because
    - @frozenhelium is busy with a few other tickets
    - @xhagrg's requested changes align with future requested work (e.g. https://github.com/NASA-IMPACT/nasa-apt-frontend/issues/406)
- #622 
    - https://github.com/NASA-IMPACT/nasa-apt/pull/638 will fully update the backend to our new PDF infrastructure. 
        - We're waiting to merge this until @frozenhelium pushes the required frontend changes to production
        - It took @sunu and @frozenhelium some back and forth to get it all working. It has been tested locally and works.
        - It will be pushed to staging very soon (hopefully Friday, 24 Feb, for further testing)
    - Once ☝️ is pushed, @frozenhelium will work on https://github.com/NASA-IMPACT/nasa-apt/issues/642, which will be ready by mid-next week.
    - We still expect to have the full new PDF architecture and all core components (table of contents, references, etc.) on production by the end of this sprint (Friday, March 3rd), given your review and approval.
- #602 
    - @sunu will have a stab at it, but it may require @frozenhelium who is our only dedicated frontend engineer (who is busy with all of the other bullets on this comment 😅 
- #636 
    - @thenav56 has completed this work. It's on staging and is awaiting the APT team's approval before we move it to production.
    - Once your team has finalized the [APT Events/Notifications spreadsheet](https://docs.google.com/spreadsheets/d/1mjEzk_Zq0ZZhPPA8EoprrabXNf7lT5QxBcdS2QXqDQc/edit#gid=0) for https://github.com/NASA-IMPACT/nasa-apt/issues/609, we can start implementing the rest of these.

cc @deborahUAH @bwbaker1 @batpad

**Comment by wrynearson** [comment]
Sorry @bwbaker1 I'm just getting this to you now:

## Mid-week update

### Sprint 3 Progress
- https://github.com/NASA-IMPACT/nasa-apt/issues/608 is in review by your side
- https://github.com/NASA-IMPACT/nasa-apt/issues/622 (Document PDFs) is ready for review by your side
    - Once approved and deployed to production, this will unblock:
        - #236
        - #552
        - #545
    - #642 is also in review by your side

### Sprint 4 Goals
- Implement new architecture for Journal PDFs, allowing for functional downloads
    - Fully implementing Journal PDFs will take more than 1 sprint (probably 2), especially with #649  
- Document PDF improvements (speeding up downloads)
- Curator dashboard error handling (#647, from #643) 
- Implement #406 

Note that next week (Mar 6-10) is a flex week for DevSeed, so expect minimal movement from us next week.

cc @deborahUAH @batpad

**Comment by wrynearson** [comment]
## Mid-week update (Sprint 4, Week 1)

### Sprint 4 progress
- Flex week this week for DevSeed, plus Holi for colleagues in India and Nepal this week.
- Misunderstanding for Document PDF requirements, adding formatting and styling tickets for this sprint.
- Investigating https://github.com/NASA-IMPACT/nasa-apt-frontend/issues/417
- Working on download performance improvements, hoping to complete this week.

### Sprint goals for next week
Document PDF export functionality is still a big blocker for users and other features (#500, etc.).

I propose that we focus on core format requirements this sprint, push them to production, and then look at the intricacies of APA/AGU reference section requirements, which will take more time to get all of those details turned into code.
- https://github.com/NASA-IMPACT/nasa-apt/issues/606#issuecomment-1457918626
- #651 (focusing on core functionality instead of _all_ edge cases this sprint)
- https://github.com/NASA-IMPACT/nasa-apt/issues/652 (alphabetical ordering)

cc @deborahUAH @bwbaker1 @batpad

**Comment by deborahUAH** [comment]
@wrynearson reminder that UAH is on spring break next week, so Brad, Camille, Ashley and I are away.  I will also be out the week after for vacation.  I have on my calendar to meet with you for both the Monday meetings so we can move things forward anyhow.
here is the [sprint review slide deck](https://docs.google.com/presentation/d/1m_xu81OHTRQN468saGROm1xtZ6QcDLgpCP7DOR8J7dc) - I filled in APT with projected completions.  Please adjust on Thursday (3/16) to indicate what got completed during the week and remove what did not and add it to the next column.

**Comment by deborahUAH** [comment]
Reminder that done = pushed to production for everything.  I agree to focus on core requirements (PDF especially) and push to production when testing shows it works.  AGU style for references is agreeable (you asked on another ticket for confirmation).   
Talk to you on Monday.

**Comment by deborahUAH** [comment]
btw, I see that curator dashboard is working again in Prod so I added it to the sprint slide.

**Comment by wrynearson** [comment]
## Mid-week update (NASA Sprint 5, Week 1)

### Sprint Goals

- [ ] Document PDF on prod in "base" state (not fully accounting for all AGU reference and styling requirements
- [ ] #608 on prod
- [ ] https://github.com/NASA-IMPACT/nasa-apt-frontend/issues/427 once videos are ready
- [ ] _Stretch goal_: Bug fixes
- [ ] _Stretch goal_: Start on Journal PDF prereq work

### Updates

- New GH status `Ready for Production Deployment` to mark everything that's been reviewed and approved by both sides.
- Document PDF work is ready to deploy to production (minus https://github.com/NASA-IMPACT/nasa-apt-frontend/issues/417 after some missing content was found in review by @bwbaker1 )
    - https://github.com/NASA-IMPACT/nasa-apt/issues/665
- https://github.com/NASA-IMPACT/nasa-apt/issues/608
    - For our initial time having this on prod, we recommend giving users the ability to delay adding MFA to their accounts just in case users run into issues. This will give us time to get user feedback before the NGAP transition
- #647 has gone through one round of testing, and will be further tested 

cc: @deborahUAH @bwbaker1 @xhagrg @batpad

**Comment by wrynearson** [comment]
# Weekly Sync Meeting
## DevSeed Updates 
- Sprint goals are on track
    - Still investigating https://github.com/NASA-IMPACT/nasa-apt-frontend/issues/417
- https://github.com/NASA-IMPACT/nasa-apt/issues/665
- Bug fixes

## IMPACT Updates
- @deborahUAH and team will meet with IMPACT leadership and stakeholders today to talk about future goals, priorities, etc.

### Priorities for next quarter
- APT in functional state is top priority
    - @deborahUAH – A user can register, create a doc, write a doc quickly, getting someone to review, then publish on APT (within a month period) without stalling, breakage, etc.
- Some users are currently using templates (Word docs and Google Docs) to write ATBDs (as recently instructed by management)
    - Importing this content from templates somehow...
        - Users export PDFs from those templates and add them to APT with some basic metadata
        - Some more complex import process
    - ... is deemed important to this definition of APT being in a functional state by @deborahUAH
    - @wrynearson flagged that this could vary widely in complexity, and could lead to many "edge cases" (certain content, formatting, styling, etc. not being supported), and that it could go against the vision of the APT (to have users edit and review within APT.
- Journal PDF is of medium priority
- Importing existing ATBDs is a lower priority

## Actions
- @wrynearson+DevSeed and IMPACT team to have prioritization and scoping exercise for FY23Q3 during next Monday's meeting 

cc @batpad

**Comment by wrynearson** [comment]
# Weekly Update – FY22Q3 Q1 S1W1

- @wrynearson working with DS designers to make sure we're implementing a new workflow for document upload that is user friendly and as simple as possible.
    - They will have designs/comments ready to review by the end of the sprint.
    - https://github.com/NASA-IMPACT/nasa-apt/issues/690 and https://github.com/NASA-IMPACT/nasa-apt/issues/675 will start once ☝️ is ready
- https://github.com/NASA-IMPACT/nasa-apt/issues/696 is resolved for one of the two PDFs in question.
    - We think the other is related to the size of the generated PDF hitting a limitation with our Lambda infrastructure (_ticket coming soon_)  
    - If so, this will unfortunately take time to fix if rearchitecting is needed
    - In the meantime, users can append `/pdf-preview` to the URL and use the browser's print functionality
- https://github.com/NASA-IMPACT/nasa-apt/issues/668 if mostly complete
- https://github.com/NASA-IMPACT/nasa-apt/pull/699 will be merged shortly, which should fix https://github.com/NASA-IMPACT/nasa-apt/issues/500#issuecomment-1502034106
- https://github.com/NASA-IMPACT/nasa-apt/issues/609 is in progress
- We can add help videos to tooltips once they are ready
    - https://github.com/NASA-IMPACT/nasa-apt/issues/615
    - https://github.com/NASA-IMPACT/nasa-apt/issues/617
cc @bwbaker1 @deborahUAH @xhagrg @batpad

**Comment by bwbaker1** [comment]
### IMPACT Weekly Update - FY23Q3 S1W2

- https://github.com/NASA-IMPACT/nasa-apt/issues/675 This is nearly complete. Impact team is currently iterating over the content and will have it ready by COB 4/28. Will send to DevSeed for review asap.
- https://github.com/NASA-IMPACT/nasa-apt/issues/673 The template is complete. We just need to add a brief file with user instructions (currently instructions are within the latex comments). We need to discuss how he will present this template for user download on APT (currently is a zipped file in 4 files inside). 
- https://github.com/NASA-IMPACT/nasa-apt/issues/692 The draft form is complete. We are still iterating over what information is necessary to collect from users. Will deliver to DevSeed for review by COB 5/1.
- https://github.com/NASA-IMPACT/nasa-apt/issues/530 Lots of progress made updating the user guide. Updating will be blocked soon until all the updates are made to the APT interface.
- https://github.com/NASA-IMPACT/nasa-apt/issues/672 This document has been updated (it was very out of date). Team is reviewing/editing now. Hopefully it will be ready early next sprint.
- https://github.com/NASA-IMPACT/nasa-apt/issues/695 @deborahUAH presented APT to the Atmospheric Observing System (AOS) team upon their request. Presentation went very well, and we already have one AOS lead checking out the interface now. AOS now discussing if they want to use APT for upcoming ATBDs.
- https://github.com/NASA-IMPACT/nasa-apt/issues/691 is started. 
-   https://github.com/NASA-IMPACT/nasa-apt/issues/533 Testing anything as necessary.
-  https://github.com/NASA-IMPACT/nasa-apt/issues/617 and https://github.com/NASA-IMPACT/nasa-apt/issues/615 were deprioritized this sprint. These are mostly complete and will have them ready S2W1.

cc @wrynearson @deborahUAH

**Comment by wrynearson** [comment]
## DevSeed Sprint Update – FY23Q3 S2

### Goals
- PDF upload workflow:
    - Finalize new PDF upload workflow with designers, approve with developers
    - Work with @bwbaker1 and team to agree on new workflow
    - Implement initial workflow changes (https://github.com/NASA-IMPACT/nasa-apt/issues/690, https://github.com/NASA-IMPACT/nasa-apt/issues/708, or what is agreed upon)
- https://github.com/NASA-IMPACT/nasa-apt/issues/594
- https://github.com/NASA-IMPACT/nasa-apt/issues/609
- https://github.com/NASA-IMPACT/nasa-apt/issues/710
- Clarify if https://github.com/NASA-IMPACT/nasa-apt/issues/500#issuecomment-1502034106 is ready for production

### Nice to have, but probably won't accomplish this sprint
- https://github.com/NASA-IMPACT/nasa-apt/issues/682 (_deprioritized due to PDF upload workflow_)
- https://github.com/NASA-IMPACT/nasa-apt-frontend/issues/471
- https://github.com/NASA-IMPACT/nasa-apt/issues/578
- https://github.com/NASA-IMPACT/nasa-apt/issues/481#issuecomment-1529115509
- https://github.com/NASA-IMPACT/nasa-apt/issues/510

cc @batpad

**Comment by wrynearson** [comment]
Update from PDF upload design call:

https://github.com/NASA-IMPACT/nasa-apt/issues/714#issuecomment-1531677802

**Comment by bwbaker1** [comment]
### IMPACT Weekly Update - FY23Q3 S2W1


- https://github.com/NASA-IMPACT/nasa-apt/issues/673 Nearly done. We need to discuss how he will present this template for user download on APT (currently is a zipped file in 4 files inside). 
- https://github.com/NASA-IMPACT/nasa-apt/issues/530 Continued working towards updating the user guide. Updating will be blocked soon until all the updates are made to the APT interface.
- https://github.com/NASA-IMPACT/nasa-apt/issues/672 . Team is reviewing/editing now. 
-   https://github.com/NASA-IMPACT/nasa-apt/issues/533 Tested a few items (new edit button, 
-  https://github.com/NASA-IMPACT/nasa-apt/issues/617 and https://github.com/NASA-IMPACT/nasa-apt/issues/615. Finishing these up. Started outlining https://github.com/NASA-IMPACT/nasa-apt/issues/616 and https://github.com/NASA-IMPACT/nasa-apt/issues/624.
- https://github.com/NASA-IMPACT/nasa-apt/issues/716 Submitted abstract/Notice of intent to the MSFC poster jamboree. https://github.com/NASA-IMPACT/nasa-apt/issues/723 Draft poster started.
- https://github.com/NASA-IMPACT/nasa-apt/issues/717 completed.

cc @wrynearson

**Comment by wrynearson** [comment]
# DevSeed Weekly Update – FY23Q3 S2W2
_Note that DevSeed will have a flex week next week. @wrynearson will be out Wednesday-Friday next week_

## Progress update (this sprint)

### We'll deploy the following to production in a few hours:

- [x] https://github.com/NASA-IMPACT/nasa-apt-frontend/pull/474
    - Addresses 
        - https://github.com/NASA-IMPACT/nasa-apt/issues/690
        - https://github.com/NASA-IMPACT/nasa-apt/issues/675
    - Ready for staging and production deployment
        - Waiting on https://github.com/NASA-IMPACT/nasa-apt/issues/674
            - _@bwbaker1 should we deploy before the video is ready, keeping the placeholder there?_
- [x] https://github.com/NASA-IMPACT/nasa-apt/issues/594
- [x] https://github.com/NASA-IMPACT/nasa-apt/issues/500#issuecomment-1502034106

### Ready for testing by IMPACT:

- [ ] https://github.com/NASA-IMPACT/nasa-apt/issues/609
- [x] https://github.com/NASA-IMPACT/nasa-apt/issues/510
- [x] https://github.com/NASA-IMPACT/nasa-apt-frontend/pull/481 (not a sprint goal)

### In progress

- [x] https://github.com/NASA-IMPACT/nasa-apt/issues/710
- [ ] https://github.com/NASA-IMPACT/nasa-apt-frontend/issues/471
- [ ] https://github.com/NASA-IMPACT/nasa-apt/pull/703
- [ ] https://github.com/NASA-IMPACT/nasa-apt/issues/727

### Blocked

- [ ] https://github.com/NASA-IMPACT/nasa-apt/issues/578
    - @naomatheus has not received permissions approval to be able to fix this.

### Deprioritized

- [ ] https://github.com/NASA-IMPACT/nasa-apt/issues/682

## Goals for next sprint

- [ ] https://github.com/NASA-IMPACT/nasa-apt/issues/727
- [ ] https://github.com/NASA-IMPACT/nasa-apt/issues/725 (this should take about 1 sprint)
- [ ] https://github.com/NASA-IMPACT/nasa-apt/issues/726 (this will take more than 1 sprint and will require sub-tickets)

cc @batpad @bwbaker1

**Comment by wrynearson** [comment]
# DevSeed Weekly Update – FY23Q3 S3W2

_Last week was a flex-week for DevSeed_

## Sprint Goals
- [ ] https://github.com/NASA-IMPACT/nasa-apt/issues/727
- [x] https://github.com/NASA-IMPACT/nasa-apt/issues/725 (on staging)
- [ ] Progress on https://github.com/NASA-IMPACT/nasa-apt/issues/726 with:
    - [x] https://github.com/NASA-IMPACT/nasa-apt/issues/733 (cancelled for now)
    - [ ] https://github.com/NASA-IMPACT/nasa-apt/pull/734 (in progress)
    - [ ] https://github.com/NASA-IMPACT/nasa-apt/pull/735 (in progress)
    - [ ] https://github.com/NASA-IMPACT/nasa-apt/pull/736 (in progress)

## Additional ongoing work

- [ ] https://github.com/NASA-IMPACT/nasa-apt-frontend/issues/471

## Blocked

- [x] https://github.com/NASA-IMPACT/nasa-apt/issues/578

## Goals for next sprint

- [ ] Finish https://github.com/NASA-IMPACT/nasa-apt/issues/726 with:
    - [ ] https://github.com/NASA-IMPACT/nasa-apt/issues/737
    - [ ] https://github.com/NASA-IMPACT/nasa-apt/issues/738

cc: @bwbaker1 @batpad

**Comment by wrynearson** [comment]
# DevSeed Update: FY23Q3S4

## Sprint Goals:

- [ ] Finish https://github.com/NASA-IMPACT/nasa-apt/issues/726
    - We ran into some issues on this, so we're now expecting to finish https://github.com/NASA-IMPACT/nasa-apt/issues/734, https://github.com/NASA-IMPACT/nasa-apt/issues/735 and https://github.com/NASA-IMPACT/nasa-apt/issues/736 by the end of this week (June 2nd).
    - Once ☝️ are done, continue working on https://github.com/NASA-IMPACT/nasa-apt/issues/737 and https://github.com/NASA-IMPACT/nasa-apt/issues/738

## Additional ongoing work
- Continue porting tests to https://github.com/NASA-IMPACT/nasa-apt-frontend/issues/471

**Comment by wrynearson** [comment]
# DevSeed Mid-Week Update: FY23Q3 S4W1

## Goals

- [ ] Getting closer on https://github.com/NASA-IMPACT/nasa-apt/issues/726, still expecting to have this on completion by the end of the sprint
    - Due to some technical reasons, we're going to ask users to decide between PDF upload and APT edit workflows at the time of ATBD creation, and imagine to explain to users the differences at this point. We can demo this to explain this better.
- [x] https://github.com/NASA-IMPACT/nasa-apt-frontend/issues/471 is in review – this was a _big_ PR
- [x] DevSeed's work on https://github.com/NASA-IMPACT/nasa-apt/issues/578 is finished, now with IMPACT

## If there's time

- [x] https://github.com/NASA-IMPACT/nasa-apt/issues/744
- [ ] https://github.com/NASA-IMPACT/nasa-apt-frontend/issues/476

cc: @bwbaker1

**Comment by wrynearson** [comment]
# DevSeed-IMPACT Weekly Check-in: FY23Q3 S4W2

## DevSeed Updates:
- https://github.com/NASA-IMPACT/nasa-apt/issues/726 is going well, still expecting to be completed by the end of this week
    - @wrynearson shared a demo of the progress so far as a video to @bwbaker1
- We'll look into email issues this week (@sunu)
- https://github.com/NASA-IMPACT/nasa-apt-frontend/issues/471 are mostly done, continuing to port tests over and improve them

## IMPACT Updates:
- Ashley working on feedback survey with Deborah and the team
- User guides have and continue to be updated

**Comment by wrynearson** [comment]
# DevSeed Mid-Week Update: FY23Q3 S4W2

## Goals

- [x] https://github.com/NASA-IMPACT/nasa-apt/issues/726 on staging, ready for review
    - We've identified a few changes and improvements to make, which are marked in the `Improvements` section of this ticket ☝️ 
- [x] Waiting to get logs for email notifications
- [ ] https://github.com/NASA-IMPACT/nasa-apt-frontend/issues/471 are mostly done, continuing to port tests over and improve them. https://github.com/NASA-IMPACT/nasa-apt-frontend/pull/466 is in review internally.

## Extras 🎉 

- [x] https://github.com/NASA-IMPACT/nasa-apt-frontend/pull/488 on staging
- [x] https://github.com/NASA-IMPACT/nasa-apt-frontend/pull/487 on staging
- [x] https://github.com/NASA-IMPACT/nasa-apt-frontend/pull/485 on staging
- [ ] Several backend improvements

cc @batpad @bwbaker1

**Comment by wrynearson** [comment]
# DevSeed-IMPACT Weekly Check-in: FY23Q3 S5W1

## DevSeed Sprint Goal
- [ ] Finish improvements in https://github.com/NASA-IMPACT/nasa-apt/issues/726 this sprint so we can push everything to production (cc @batpad @frozenhelium)
    - Still need to scope out https://github.com/NASA-IMPACT/nasa-apt/issues/753

## IMPACT Sprint Goals
- Work through this quarter's backlog

cc @bwbaker1

**Comment by wrynearson** [comment]
# DevSeed Update (FY23Q3 S5W2)

- Scheduling next prod release for this Wednesday/Thursday
- https://github.com/NASA-IMPACT/nasa-apt/issues/726
    - All "Improvements" are on staging, ready for production deployment later this week
    - We're investigating issues related to https://github.com/NASA-IMPACT/nasa-apt-frontend/issues/495 on the published document page, but we're having a hard time reproducing the issue @bwbaker1 
- Next, we'll look into https://github.com/NASA-IMPACT/nasa-apt/issues/753, but it won't be ready in time for the production release later this week

cc @bwbaker1

**Comment by wrynearson** [comment]
# DevSeed Update (FY23Q4)
- @wrynearson was out for PI planning
- @bwbaker1 out until today
- @wrynearson updated the APT project board
- DevSeed met today to review the PI objectives. Everything looks good from us 👍

**Comment by wrynearson** [comment]
# DevSeed Update – FY23Q4S1 W1
- Focus on #649 and scoping/implementing journal PDF
- Various other small fixes, tweaks, etc. (see GH Project for full list)

**Comment by wrynearson** [comment]
# DevSeed Update – FY23Q4S1 W2

## Complete

- [x] https://github.com/NASA-IMPACT/nasa-apt/issues/578
- [x] https://github.com/NASA-IMPACT/nasa-apt/issues/664
- [x] https://github.com/NASA-IMPACT/nasa-apt/issues/682
- [x] https://github.com/NASA-IMPACT/nasa-apt/issues/731
- [x] https://github.com/NASA-IMPACT/nasa-apt/issues/740
- [x] https://github.com/NASA-IMPACT/nasa-apt/issues/770
- [x] https://github.com/NASA-IMPACT/nasa-apt/issues/780

## In Progress/Review

- [ ] https://github.com/NASA-IMPACT/nasa-apt/issues/781 (code complete, in review by IMPACT)
- [ ] https://github.com/NASA-IMPACT/nasa-apt/issues/586 (initial code complete, adding additional functionality)
- [ ] https://github.com/NASA-IMPACT/nasa-apt/issues/753 (in review)
- [ ] https://github.com/NASA-IMPACT/nasa-apt-frontend/issues/514 (frontend is complete, wrapping up backend changes)
- [ ] https://github.com/NASA-IMPACT/nasa-apt/issues/761 (in review)

**Comment by wrynearson** [comment]
# DevSeed Update – FY23Q4S2 W1

## Sprint Goals

- [ ] Finish https://github.com/NASA-IMPACT/nasa-apt-frontend/issues/514
- [ ] Finish https://github.com/NASA-IMPACT/nasa-apt/issues/586
- [ ] Finish https://github.com/NASA-IMPACT/nasa-apt/issues/753
- [ ] https://github.com/NASA-IMPACT/nasa-apt/issues/548
- [ ] Start on https://github.com/NASA-IMPACT/nasa-apt-frontend/issues/421
- [ ] https://github.com/NASA-IMPACT/nasa-apt/issues/681
- [ ] Start on https://github.com/NASA-IMPACT/nasa-apt/issues/756
- [ ] https://github.com/NASA-IMPACT/nasa-apt/issues/779
- [ ] Come up with a plan for https://github.com/NASA-IMPACT/nasa-apt/issues/784 and https://github.com/NASA-IMPACT/nasa-apt/issues/785
- [ ] Working first version of https://github.com/NASA-IMPACT/nasa-apt/issues/786
- [ ] https://github.com/NASA-IMPACT/nasa-apt/issues/493

cc @bwbaker1 @batpad @sunu @kamicut @thenav56 @frozenhelium

**Comment by wrynearson** [comment]
# DevSeed Mid-Week Update – FY23Q4S2 W1:

- https://github.com/NASA-IMPACT/nasa-apt-frontend/issues/514 is ready for IMPACT review
- https://github.com/NASA-IMPACT/nasa-apt/issues/586 is ready for IMPACT review
- https://github.com/NASA-IMPACT/nasa-apt/issues/753 is ready for IMPACT review
- https://github.com/NASA-IMPACT/nasa-apt/issues/493 is on staging
- Great progress on https://github.com/NASA-IMPACT/nasa-apt/issues/786 – we now have a working "first draft". We'll continue to test
- https://github.com/NASA-IMPACT/nasa-apt/issues/756 will be simple to implement once we receive the text

**Comment by wrynearson** [comment]
## DevSeed Mid-Week Update – FY23Q4S2 W2:

### In Review by IMPACT

- [ ] https://github.com/NASA-IMPACT/nasa-apt/issues/493
- [ ] https://github.com/NASA-IMPACT/nasa-apt/issues/548

### In Review by DevSeed

- [ ] https://github.com/NASA-IMPACT/nasa-apt/issues/786 (waiting to merge until we finish https://github.com/NASA-IMPACT/nasa-apt-frontend/issues/421)

### In Progress
- https://github.com/NASA-IMPACT/nasa-apt-frontend/issues/421 (just about done)
- Looking into reference importation and management
     - https://github.com/NASA-IMPACT/nasa-apt-frontend/issues/421
     - Reference management is tricky, and expected by users.
     - tl;dr – we think we can manage _all_ styling and formatting requirements for AGU through bibtex uploads. This assumes that users use external reference managers to manage their references.
     - The easiest way to manage this would be to remove the APT reference editor interface, and rely on external reference management. This is being discussed
     - We're waiting to implement https://github.com/NASA-IMPACT/nasa-apt/issues/784 and https://github.com/NASA-IMPACT/nasa-apt/issues/785 until we determine the next steps for references 
 
### Blocked
- https://github.com/NASA-IMPACT/nasa-apt/issues/756 (waiting for updated text, moved to sprint 3)

### Moved to Sprint 3
- https://github.com/NASA-IMPACT/nasa-apt/issues/779
- https://github.com/NASA-IMPACT/nasa-apt/issues/761#issuecomment-1679084516 (adding reviewer information to the upload view mode)

cc @bwbaker1 @batpad @sunu @thenav56 @kamicut @frozenhelium

**Comment by wrynearson** [comment]
# DevSeed Mid-Week Update: FY23Q4S3W2:

## Done (on prod)
- Elastic Search is now encrypted on staging and production (cc @ChrisPhillips1024 )
- https://github.com/NASA-IMPACT/nasa-apt/issues/548
- https://github.com/NASA-IMPACT/nasa-apt/issues/493
- https://github.com/NASA-IMPACT/nasa-apt/issues/781

## In Review (IMPACT)
- https://github.com/NASA-IMPACT/nasa-apt/issues/756
- https://github.com/NASA-IMPACT/nasa-apt-frontend/issues/421 (also in review by DevSeed)
- https://github.com/NASA-IMPACT/nasa-apt/issues/790

## In Progress
- https://github.com/NASA-IMPACT/nasa-apt/issues/779
- Reference work:
    - https://github.com/NASA-IMPACT/nasa-apt-frontend/issues/425
    - https://github.com/NASA-IMPACT/nasa-apt/issues/784
    - https://github.com/NASA-IMPACT/nasa-apt/issues/785 

## Deprioritized (for next sprint):
- https://github.com/NASA-IMPACT/nasa-apt/issues/681
- https://github.com/NASA-IMPACT/nasa-apt/issues/753 (reviewer info in PDF upload view mode – Journal PDF took priority)

## Sprint 4 Goals:
- [ ] Journal PDF (styling and line numbers) on production
- [ ] https://github.com/NASA-IMPACT/nasa-apt/issues/807 / https://github.com/NASA-IMPACT/nasa-apt/issues/798
- [ ] Act on references discussion

## Admin
- Most of the DevSeed team will be at a team retreat next week (Sep 4-8). Expect near-zero development progress, but please reach out for critical issues.
- @wrynearson will be out from Sep 11 - 22. Several other team members will be as well, but we'll set up a way to keep sharing progress, etc.
- @wrynearson and @bwbaker1 should document the definition of the "maintenance mode" we're going into for next FY to make sure IMPACT and DevSeed are on the same page about everything.

cc @bwbaker1 @batpad @sunu @kamicut @frozenhelium @thenav56 @aboydnw

**Comment by bwbaker1** [comment]
@kamicut @frozenhelium Will one or both of you provide a brief update on the work that is in progress? If possible, please provide this by Sept. 12th. This information will be useful for a meeting we have planned on Sept. 13th. Thanks in advance!

**Comment by kamicut** [comment]
Hey @bwbaker1 here is an our update from @kamicut and @frozenhelium 

## DevSeed Mid-Week Update FY23Q4S4W2

## Done
- https://github.com/NASA-IMPACT/nasa-apt/issues/790

## In Review (IMPACT)
- https://github.com/NASA-IMPACT/nasa-apt/issues/800

## In Review (Devseed)
- https://github.com/NASA-IMPACT/nasa-apt/issues/779

## In progress
- https://github.com/NASA-IMPACT/nasa-apt/issues/681
- https://github.com/NASA-IMPACT/nasa-apt-frontend/issues/421
- https://github.com/NASA-IMPACT/nasa-apt-frontend/issues/514
- https://github.com/NASA-IMPACT/nasa-apt/issues/801

I'm going to go wait for @wrynearson to come back to set goals for sprint 5

**Comment by bwbaker1** [comment]
@kamicut Thank you!

**Comment by bwbaker1** [comment]
@kamicut Please provide another progress update this week. Thanks!

**Comment by kamicut** [comment]
Hey @bwbaker1, we'll post an update after our team check-in on Thursday if that's alright with you!

**Comment by bwbaker1** [comment]
@kamicut That is fine. Thanks!

**Comment by kamicut** [comment]
## DevSeed Mid-Week Update FY23Q4S5W1

## In Review (IMPACT)
- https://github.com/NASA-IMPACT/nasa-apt/issues/800

## In Review (Devseed)
- https://github.com/NASA-IMPACT/nasa-apt/issues/779
- https://github.com/NASA-IMPACT/nasa-apt-frontend/issues/421
- https://github.com/NASA-IMPACT/nasa-apt-frontend/issues/514


## In progress
- https://github.com/NASA-IMPACT/nasa-apt/issues/681
- https://github.com/NASA-IMPACT/nasa-apt/issues/789
- https://github.com/NASA-IMPACT/nasa-apt/issues/795 and https://github.com/NASA-IMPACT/nasa-apt-frontend/issues/463 (related to disabling MFA locally)

## Blocked
- https://github.com/NASA-IMPACT/nasa-apt/issues/801 (Waiting for @wrynearson)

**Comment by bwbaker1** [comment]
Thank you @kamicut!

**Comment by wrynearson** [comment]
# DevSeed Mid-Week Update FY23Q4 S5W2

## To-do
- Prioritizing feedback from https://github.com/NASA-IMPACT/nasa-apt-frontend/issues/421 
- https://github.com/NASA-IMPACT/nasa-apt/issues/801

## In Progress
- https://github.com/NASA-IMPACT/nasa-apt/issues/681
- https://github.com/NASA-IMPACT/nasa-apt/issues/789
- https://github.com/NASA-IMPACT/nasa-apt/issues/795
- https://github.com/NASA-IMPACT/nasa-apt-frontend/issues/463

## In Review (DevSeed)
- https://github.com/NASA-IMPACT/nasa-apt-frontend/pull/546

## In Review (IMPACT)
- https://github.com/NASA-IMPACT/nasa-apt/issues/779 (we added more text in the error message)

## To deploy to prod
- https://github.com/NASA-IMPACT/nasa-apt-frontend/issues/514

cc @bwbaker1 @kamicut @sunu @thenav56
